lib/rucio/common/types.py

```diff
@@ -236,10 +236,10 @@ class TokenValidationDict(TypedDict):
 
 class IPDict(TypedDict):
     ip: str
```

Comment:
Is there a reason for the IP to be required? There's a lot of `ignore`s further down the chain bc of ip

Reply:
Good point; I've now split this type into two types:
- `IPWithLocationDict` is the one returned by `detect_client_location`, which always returns `ip`, `fqdn` and `site` as `str`
- `IPDict` is used in all other contexts, where any of those values might be `None`